homeassistant/components/enphase_envoy/binary_sensor.py

```diff
@@ -0,0 +1,142 @@
+"""Support for Enphase Envoy solar energy monitor."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from pyenphase import (
+    EnvoyData,
+    EnvoyEncharge,
+)
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+)
+
+from .const import DOMAIN
+from .coordinator import EnphaseUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class EnvoyEnchargeRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[EnvoyEncharge], bool]
+
+
+@dataclass
+class EnvoyEnchargeBinarySensorEntityDescription(
+    BinarySensorEntityDescription, EnvoyEnchargeRequiredKeysMixin
+):
+    """Describes an Envoy Encharge sensor entity."""
+
+
+ENCHARGE_SENSORS = (
+    EnvoyEnchargeBinarySensorEntityDescription(
+        key="communicating",
+        translation_key="communicating",
+        device_class=BinarySensorDeviceClass.CONNECTIVITY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda encharge: encharge.communicating,
+    ),
+    EnvoyEnchargeBinarySensorEntityDescription(
+        key="dc_switch",
+        translation_key="dc_switch",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda encharge: not encharge.dc_switch_off,
+    ),
+    EnvoyEnchargeBinarySensorEntityDescription(
+        key="operating",
+        translation_key="operating",
```

Comment:
```suggestion
        device_class=BinarySensorDeviceClass.RUNNING,
```
Would this be a good fit for this device class?

Reply:
I went back and forth on that one. To my knowledge `operating` from the API means the battery is on but may or may be idle (not charging or discharging). `Running` feels like its actively charging or discharging to me, but definitely open to changing it